politeiawww/codestats.go

```diff
@@ -0,0 +1,298 @@
+	// Run until start date is after end date, it's incremented by a month
+	// a the end of the loop.
+	for startDate.After(endDate) {
+		month := startDate.Month()
+		year := startDate.Year()
+		cu := user.CMSCodeStatsByUserMonthYear{
+			GithubName: requestedUser.GitHubName,
+			Month:      int(month),
+			Year:       year,
+		}
+		payload, err := user.EncodeCMSCodeStatsByUserMonthYear(cu)
+		if err != nil {
+			return nil, err
+		}
+		pc := user.PluginCommand{
+			ID:      user.CMSPluginID,
+			Command: user.CmdCMSCodeStatsByUserMonthYear,
+			Payload: string(payload),
+		}
+
+		// Execute plugin command
+		pcr, err := p.db.PluginExec(pc)
+		if err != nil {
+			return nil, err
+		}
+
+		// Decode reply
+		reply, err := user.DecodeCMSCodeStatsByUserMonthYearReply(
+			[]byte(pcr.Payload))
+		if err != nil {
+			return nil, err
+		}
+		allRepoStats = append(allRepoStats,
+			convertCodeStatsFromDatabase(reply.UserCodeStats)...)
+
+		startDate = time.Date(startDate.Year(), startDate.Month()+1,
+			startDate.Day(), startDate.Hour(), startDate.Minute(), 0, 0,
+			time.UTC)
+	}
+	return &cms.UserCodeStatsReply{
+		RepoStats: allRepoStats,
+	}, nil
+}
+
+func (p *politeiawww) updateCodeStats(org string, repos []string, start, end int64) error {
+
+	// make sure tracker was created, if not alert for them to check github api
+	// token config
+	if p.tracker == nil {
+		return fmt.Errorf("code tracker not running")
+	}
+
+	//p.tracker.Update(org, repos, start, end)
+
+	// Go fetch all Development contractors to update their stats
+	cu := user.CMSUsersByDomain{
+		Domain: int(cms.DomainTypeDeveloper),
+	}
+	payload, err := user.EncodeCMSUsersByDomain(cu)
+	if err != nil {
+		return err
+	}
+	pc := user.PluginCommand{
+		ID:      user.CMSPluginID,
+		Command: user.CmdCMSUsersByDomain,
+		Payload: string(payload),
+	}
+
+	// Execute plugin command
+	pcr, err := p.db.PluginExec(pc)
+	if err != nil {
+		return err
+	}
+
+	// Decode reply
+	reply, err := user.DecodeCMSUsersByDomainReply([]byte(pcr.Payload))
+	if err != nil {
+		return err
+	}
+
+	currentMonth := int(time.Now().Month())
+	currentYear := time.Now().Year()
+
+	for _, u := range reply.Users {
+		if u.GitHubName == "" {
+			// Just move along since user has no github name set
+			continue
+		}
+
+		cu := user.CMSCodeStatsByUserMonthYear{
+			GithubName: u.GitHubName,
+			Month:      currentMonth,
+			Year:       currentYear,
+		}
+		payload, err := user.EncodeCMSCodeStatsByUserMonthYear(cu)
+		if err != nil {
+			return err
+		}
+		pc := user.PluginCommand{
+			ID:      user.CMSPluginID,
+			Command: user.CmdCMSCodeStatsByUserMonthYear,
+			Payload: string(payload),
+		}
+
+		// Execute plugin command
+		pcr, err := p.db.PluginExec(pc)
+		if err != nil {
+			return err
+		}
+
+		// Decode reply
+		reply, err := user.DecodeCMSCodeStatsByUserMonthYearReply(
+			[]byte(pcr.Payload))
+		if err != nil {
+			return err
+		}
+
+		githubUserInfo, err := p.tracker.UserInfo(org,
```

Comment:
Why does this route requests data from the code tracker?  Shouldn't it just be requesting data from the userdb?